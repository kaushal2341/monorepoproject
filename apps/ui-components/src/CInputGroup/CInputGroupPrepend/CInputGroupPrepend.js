import React, {memo} from 'react'
import {InputGroup} from 'react-bootstrap';

const CInputGroupPrepend = ({id, classes, prepend}) => {
    return (
        <InputGroup.Prepend
            id={id}
            className={classes}>
            {prepend && prepend.map((pp, index) => pp)}
        </InputGroup.Prepend>
    )
}
export default memo(CInputGroupPrepend);