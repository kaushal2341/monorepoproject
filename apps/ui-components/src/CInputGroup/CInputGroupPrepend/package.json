{
    "name": "@cogent/c-inputgroup-prepend",
    "main": "CInputGroupPrepend.js",
    "version": "0.0.0",
    "license": "MIT",
    "private": true
}