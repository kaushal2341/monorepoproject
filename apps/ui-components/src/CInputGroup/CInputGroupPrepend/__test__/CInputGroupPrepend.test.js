import CInputGroupPrepend from '../CInputGroupPrepend';
import {InputGroup, Button} from 'react-bootstrap';

expect.addSnapshotSerializer(enzymeSerializer);
describe('CInputGroup Prepend', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(
            <CInputGroupPrepend
                id="prependInputGroup"
                classes="hello"
                append={''}/>);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('should render component', () => {
        expect(wrapper).toBeDefined();
    });

    it('should have given props available', () => {
        const propsForAppend = ['id', 'classes', 'append'];
        const appendPropsAv = Object.keys(wrapper.props())
        propsForAppend.map((propsForApp, i) => {
            expect(propsForAppend).toContain(appendPropsAv[i])
        });
    });

    it('should have atleast one element of input inside append element', () => {
        const p = [<InputGroup.Text key="addOn-1" id="basic-addOn3">
            https://example.com/users/
        </InputGroup.Text>,
            <Button key="button2" id="button2" variant="outline-secondary">Button</Button>
        ]
        wrapper.setProps({prepend: p})
        expect(wrapper.find('#basic-addOn3').length).toBeGreaterThan(1);
    });

    describe('Snapshot testing', () => {
        it('should match the previous snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
})