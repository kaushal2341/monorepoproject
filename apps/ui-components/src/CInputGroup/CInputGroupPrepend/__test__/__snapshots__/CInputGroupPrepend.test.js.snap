// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CInputGroup Prepend Snapshot testing should match the previous snapshot 1`] = `
<Memo(CInputGroupPrepend)
  append=""
  classes="hello"
  id="prependInputGroup"
>
  <InputGroupPrepend
    className="hello"
    id="prependInputGroup"
  >
    <div
      className="hello input-group-prepend"
      id="prependInputGroup"
    />
  </InputGroupPrepend>
</Memo(CInputGroupPrepend)>
`;
